### ISSUE TITLE
darwinModules

### ISSUE THREAD

**Issue by aspauldingcode** [Gerg-L/spicetify-nix#243]
Please make a darwin module.

I can use home-manager, yes.
It would be nice to have system-wide nix-darwin module! :)

**Comment by Gerg-L** [comment]
Can you try the nixosModule, it might just work

**Comment by Gerg-L** [comment]
i don't have a darwin machine so i can't do any testing on darwin, someone try the nixosModule and let me know if it breaks

**Comment by aspauldingcode** [comment]
It rebuilds fine. 
however, I’m not sure how to test because my spotify won’t uninstall, and it was updated with a micro update which destroyed the spicetify from homemanager I was using.

**Comment by aspauldingcode** [comment]
I’m doing a fresh install to test this. Think I had corrupted nix store.

**Comment by MillerApps** [comment]
Also breaks on update for me as well. Doing a Darwin rebuild doesn't fix it. I'm very new to Nix as a whole, so I will be following for updates here.

**Comment by gigamonster256** [comment]
The nixosModule works fine for me on nix-darwin using the nixos-24.11 nixpkgs channel. What sort of "broken" behavior do y'all observe? I usually get a notification to update Spotify when I reboot my mac but I just ignore it.

**Comment by aspauldingcode** [comment]
<img width="1440" alt="Screenshot 2025-01-13 at 3 00 30 PM" src="https://github.com/user-attachments/assets/05fe1e67-3b0d-4625-af80-9778f1f6cc96" />

Fresh install (of macOS, nix configuration) 
spicetify-nix using nixos module on Nix-Darwin works just fine.

**Comment by aspauldingcode** [comment]
@Gerg-L 

> spicetify-nix using nixos module on Nix-Darwin works just fine.

**Comment by MillerApps** [comment]
I made the mistake of updating Spotify, it would seem.

**Comment by gigamonster256** [comment]
> I made the mistake of updating Spotify, it would seem.

~~What channel of nixpkgs are you on/what does `/Applications/Nix\ Apps/Spotify.app/Contents/MacOS/Spotify --version` output?~~ What seems to be broken?

It looks like the flake on your github is using the homeManagerModule so the path above wouldnt work, it'd be something like `$HOME/Applications/Home\ Manager\ Apps/Spotify.app/Contents/MacOS/Spotify --version` when installed through home-manager

You could always switch your darwin config to one with spicetify disabled, garbage collect the generations with spicetify enabled to make sure it's removed from the nix store, and re-enable spicetify. This should get you back to before Spotify tried to update itself.

One of the limitations of nix on mac is that the nix store is mutable by the root user unlike nixos where it's only mutable by the nix-daemon. Some mac apps will try to auto-update, fail since the location where they are installed is owned by root, but then attempt to elevate themselves to root to force an update which can screw up apps meant to be managed by nix

**Comment by MillerApps** [comment]
- Unstable  
-  Spotify version 1.2.53.440.g7b2f582a, Copyright (c) 2025, Spotify Ltd  
-  The theme/extension is no longer applied.  
This is all new to me; any help or pointers are appreciated.

**Comment by gigamonster256** [comment]
> * Spotify version 1.2.53.440.g7b2f582a, Copyright (c) 2025, Spotify Ltd

Yeah that's not from nixpkgs, try removing the app from the nix store and re-installing it. See the edited comment above for more info.

**Comment by MillerApps** [comment]
> You could always switch your darwin config to one with spicetify disabled, garbage collect the generations with spicetify enabled to make sure it's removed from the nix store, and re-enable spicetify. This should get you back to before Spotify tried to update itself.
> 

That didn't work due to me having corrupted my nix store. Had to rebuild the store but its working now. 

The command for anyone else in the same boat:
```sh
sudo nix-store --verify --check-contents --repair
```
 Thanks a million for the help @gigamonster256

**Comment by Gerg-L** [comment]
https://github.com/Gerg-L/spicetify-nix/commit/db77f79d4262f9bfb36f7cef216b64dd82383134 adds `darwinModules`
they're the same as `nixosModules` i need to do some work and separate stuff a bit better.
